cogs/live_ping.py

```diff
@@ -0,0 +1,470 @@
+    @commands.command()
+    async def test_pipeline(self, ctx, include_conference: str = "yes"):
+        if ctx.message.author.id not in [625907860861091856, 411166117084528640]:
+            await ctx.reply("you are not allowed to use this command :(")
+            return
+
+        if (include_conference := include_conference.lower().strip()) not in ["yes", "no"]:
+            await ctx.reply("Invalid argument. Please use `yes` or `no`")
+            return
+
+        await ctx.reply(
+            f"Test Notification for 1 Minute from now\n`includeConferenceDetails` -> `{include_conference.upper()}`"
+        )
+        # this doesn't need a lock? i think?
+        self._pending_notifications.insert(
+            0,
+            Notification(
+                event=Event(get_test_event_data(include_conference=(include_conference == "yes"))),
+                channel_id=ctx.channel.id,
+                time=datetime.datetime.now() + datetime.timedelta(minutes=1),
+                calendar_name="Pinger Test Calendar",
+                n_type="default",
+                icon_url="https://i.imgur.com/HJo8u3y.png",
+            ),
+        )
+        # restart the task
+        if self._task:
+            self._task.cancel()
+        self._task = self.bot.loop.create_task(self.dispatch_notifications())
+
+    @commands.command()
+    async def coming_up(self, ctx):
+        if ctx.message.author.id not in [625907860861091856, 411166117084528640]:
+            await ctx.reply("you are not allowed to use this command :(")
+            return
+        if len(self._pending_notifications) == 0:
+            await ctx.reply("No upcoming notifications")
+            return
+
+        #  partition _pendingNotifications into 5s
+        parts = [self._pending_notifications[i : i + 5] for i in range(0, len(self._pending_notifications), 5)]
+        embeds = []
+        for i in range(len(parts)):
+            e = discord.Embed(title=f"Upcoming Notifications", color=discord.Colour.blurple())
+            part = parts[i]
+            for notification in part:
+                e.add_field(
+                    name=notification.event.name + " - `" + notification.calendar_name + "`",
+                    inline=False,
+                    value=f'{discord.utils.format_dt(notification.time, "R")} {"`RMND`" if notification.type == "reminder" else "`EVNT`"}',
+                )
+            embeds.append(e)
+        await Paginator.Simple().start(ctx, embeds)
+
+
+async def setup(bot: IITMBot):
+    await bot.add_cog(LivePinger(bot))
+
+
+# Helpers
+
+
+def is_eligible(event: str):
+    return True
+    if name is None or name.strip() == "":
+        return False
+
+    name = name.strip().lower()
+
+    if not name.startswith("live"):
+        return False
+
+    # Add other negative checks below.
+
+    return True
+
+
+def extract_google_meet_links(text: str) -> list[str] | None:
+    if not text:
+        return None
+    return MEET_LINK_REGEX.findall(text)
```

Comment:
this needn't exist. See above for optimized version. `findall` evaluates eagerly and doesn't stop with the first occurrence, which is all we care about anyway.
```suggestion
```